backup: TestBackupCompactionLocAware failed

backup.TestBackupCompactionLocAware [failed](https://mesolite.cluster.engflow.com/invocations/default/c5aceaae-d68e-4f28-a6bf-70af9e27f0ea?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [33e99ded03cf98616646cab43d1edf579e49fb72](https://github.com/cockroachdb/cockroach/commits/33e99ded03cf98616646cab43d1edf579e49fb72):


```
=== RUN   TestBackupCompactionLocAware
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestBackupCompactionLocAware1236692442
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:95: cluster virtualization disabled due to issue: #142798 (expected label: C-test-failure)
    test_server_shim.go:679: DRPC is enabled (via testserver factory defaults)
    testutils.go:326: no Invalid Descriptors
    compaction_test.go:1148: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupCompactionLocAware1236692442
--- FAIL: TestBackupCompactionLocAware (95.24s)
=== RUN   TestBackupCompactionLocAware/fully-matching
    jobs_verification.go:91: job failed: failed to run backup compaction: processor in dc=dc1 attempted to read file from dc=dc2
--- FAIL: TestBackupCompactionLocAware/fully-matching (27.88s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupCompactionLocAware.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59240

backup.TestBackupCompactionLocAware [failed](https://mesolite.cluster.engflow.com/invocations/default/0790dcb0-9527-45a1-8c18-0494003bff94?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [0d609bc851df2d9332468d62a8408d2e5c3ac21e](https://github.com/cockroachdb/cockroach/commits/0d609bc851df2d9332468d62a8408d2e5c3ac21e):


```
=== RUN   TestBackupCompactionLocAware
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestBackupCompactionLocAware2053504951
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:95: cluster virtualization disabled due to issue: #142798 (expected label: C-test-failure)
    test_server_shim.go:679: DRPC is enabled (via testserver factory defaults)
    testutils.go:326: no Invalid Descriptors
    compaction_test.go:1148: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupCompactionLocAware2053504951
--- FAIL: TestBackupCompactionLocAware (64.42s)
=== RUN   TestBackupCompactionLocAware/fully-matching
    jobs_verification.go:91: job failed: failed to run backup compaction: processor in dc=dc1 attempted to read file from dc=dc2
--- FAIL: TestBackupCompactionLocAware/fully-matching (23.76s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupCompactionLocAware.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/backup/compaction_test.go
+++ b/pkg/backup/compaction_test.go
@@ -870,7 +870,7 @@ func TestBackupCompactionLocAware(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderRace(t, "too slow")
+	skip.UnderDuress(t, "too slow")
 
 	var hookFn func(processorLocality roachpb.Locality, fileLocality string) error
 	knobs := base.TestingKnobs{
@@ -913,7 +913,7 @@ func TestBackupCompactionLocAware(t *testing.T) {
 	}
 
 	const numAccounts = 1000
-	_, db, _, cleanupFn := backupRestoreTestSetupWithParams(t, 3, numAccounts, InitManualReplication, args)
+	_, db, _, cleanupFn := backupRestoreTestSetupWithParams(t, 3, numAccounts, NoInitManipulation, args)
 	defer cleanupFn()
 
 	getLatestFullDir := func(collectionURI []string) string {

